src/state/mod.rs

```diff
@@ -61,20 +60,22 @@ impl ClientState {
 /// Abstraction around the data store to avoid unnecessary request on client initialization.
 #[async_trait::async_trait]
 pub trait StateStore: Send + Sync {
-    /// Set up connections or check files exist to load/save state.
-    fn open(&self, path: &Path) -> Result<()>;
+    /// Signals to the `AsyncClient` if this is the first time a StateStore` has been used.
+    ///
+    /// Returns true if `StateStore` has been set up and ready to be loaded from.
+    async fn initial_use(&self) -> Result<bool>;
     /// Loads the state of `BaseClient` through `StateStore::Store` type.
-    async fn load_client_state(&self, path: &Path) -> Result<ClientState>;
+    async fn load_client_state(&self) -> Result<ClientState>;
     /// Load the state of a single `Room` by `RoomId`.
-    async fn load_room_state(&self, path: &Path, room_id: &RoomId) -> Result<Room>;
+    async fn load_room_state(&self, room_id: &RoomId) -> Result<Room>;
```

Comment:
I think we can get away without this one as well. The method that loads all rooms should be enough.